### PR TITLE
feat(matrix): Enhance matrix manipulation with advanced helper methods

### PR DESCRIPTION
**Pull request by mitsuki31**
## Overview

This pull request introduces several new helper functions and methods to enhance the functionality of the matrix manipulation module. These additions aim to provide more flexibility and efficiency in working with matrices, including the ability to add, insert, drop rows and columns, as well as calculate matrix minors.

## Notable Changes

- **AddColumn & AddRow:** Two new functions, `addColumn` and `addRow`, have been added to append arrays to the last column or row of the matrix, respectively, without causing shallow copies.
  - `addRow`
    ```java
    Matrix addRow(double[] a);
    static Matrix addRow(Matrix m, double[] a);
    ```
  - `addColumn`
    ```java
    Matrix addColumn(double[] a);
    static Matrix addColumn(Matrix m, double[] a);
    ```

- **InsertRow & InsertColumn:** New helper functions, `insertRow` and `insertColumn`, are introduced to insert arrays at specific row or column indices, shifting existing rows or columns accordingly.
  - `insertRow`
    ```java
    Matrix insertRow(int row, double[] a);
    static Matrix insertRow(Matrix m, int row, double[] a);
    ```
  - `insertColumn`
    ```java
    Matrix insertColumn(int col, double[] a);
    static Matrix insertColumn(Matrix m, int col, double[] a);
    ```

- **MinorMatrix:** A new method, `minorMatrix`, calculates and constructs a new matrix representing the minor of a given matrix, useful for linear algebra operations.
  ```java
  Matrix minorMatrix(int row, int col);
  static Matrix minorMatrix(int row, int col);
  ```

- **DropRow & DropColumn:** Two new methods, `dropRow` and `dropColumn`, are introduced to remove specified rows or columns from the matrix, with support for negative indexing.
  - `dropRow`
    ```java
    Matrix dropRow(int row);
    static Matrix dropRow(Matrix m, int row);
    ```
  - `dropColumn`
    ```java
    Matrix dropColumn(int col);
    static Matrix dropColumn(Matrix m, int col);
    ```

## Description

These additions significantly enhance the matrix manipulation capabilities of the `Matrix` class. With `addColumn` and `addRow`, users can conveniently append arrays to the matrix without worrying about shallow copies, which is particularly useful for large matrices where memory efficiency is crucial. The `insertRow` and `insertColumn` functions provide users with the flexibility to insert arrays at specific indices, allowing for more precise manipulation of matrix data.

Moreover, the introduction of the `minorMatrix` method adds advanced functionality to the `Matrix` class, enabling users to calculate minors of matrices efficiently. This can be particularly valuable in various linear algebra operations, such as determining cofactors and calculating determinants.

Additionally, the `dropRow` and `dropColumn` methods offer a convenient way to remove rows or columns from matrices, enhancing the module's usability and versatility. With support for negative indexing, users can easily specify rows or columns to remove relative to the end of the matrix.

Overall, these new functionalities not only enrich the matrix manipulation capabilities of the module but also improve its performance and usability, making it a more powerful tool for handling matrix operations.

## Summary

In summary, these changes adds a suite of new functionalities to the matrix manipulation, including helper functions for appending, inserting, and dropping rows and columns, as well as a method for calculating matrix minors. These additions contribute to making the `Matrix` class more versatile and efficient in handling matrix operations.

### CODE REVIEW

**Comment by mitsuki31** [comment]
## ⚠️ Deprecation

The `change` method are no longer recommended to use and has been deprecated due to inefficiency to change and update a specific row entries of the matrix. Instead, you can utilize the new method called `insertRow` to change row entries. Moreover, if you want to change a specific column of the matrix, the `insertColumn` method can help in more efficient way.

By deprecating the `change` method, the `select` method are no longer utilized and deprecated.
